### PR TITLE
Fix/dref final changes

### PR DESCRIPTION
**Pull request by puranban**

### CODE REVIEW

File: src/root/components/DrefPdfDocument/index.tsx [hunk 1]
```diff
@@ -687,9 +666,9 @@ function DrefPdfDocument(props: Props) {
         {dref?.needs_identified.length > 0 && (
           <View style={pdfStyles.niSection}>
             <Text style={pdfStyles.sectionHeading}>
-              {isImminentOnset
-                ? strings.drefExportImminentNeedsGapsIdentified
-                : strings.drefExportNeedsGapsIdentified}
+              {isImminentOnset ?
+                strings.drefFormImminentNeedsIdentified
```
Comment (AdityaKhatri) [inline src/root/components/DrefPdfDocument/index.tsx:670]: The previous formatting is good. Lets revert it

File: src/root/lang.js [hunk 2]
```diff
@@ -3359,63 +3389,64 @@ export default {
...
+  //drefExportSafetyConcerns: 'Any potential risk to the implementation of this operation? What will be the mitigation measures?  Any safety and security concerns? How will the National Society ensure the safety of the staff, volunteers and community?',
+  //drefExportPmerDescription: 'How will this operation be monitored? Will there be IFRC monitoring visits? How will it be deployed?',
+  //drefExportCommunicationDescription: 'Please briefly explain the National Societies communication strategy for this operation. Will the IFRC be supporting with communication? What roles will be involved?',
+  //drefExportPlannedIntervention: 'Planned Intervention',
+  //drefExportNationalSocietyContact: 'National Society contact:',
+  //drefExportAppealManager: 'IFRC Appeal Manager:',
+  //drefExportProjectManager: 'IFRC Project Manager:',
+  //drefExportIfrcEmergency: 'IFRC focal point for the emergency:',
+  //drefExportMediaContact: 'Media Contact:',
```
Comment (AdityaKhatri) [inline src/root/lang.js:3449]: Lets remove them if not required

File: src/root/components/OperationalUpdatePdfDocument/index.tsx [hunk 3]
```diff
@@ -530,9 +510,9 @@ function OperationalUpdatePdfDocument(props: Props) {
               {operationalUpdateResponse?.event_description && (
                 <View style={pdfStyles.subSection}>
                   <Text style={pdfStyles.subSectionHeading}>
-                    {isImminentOnset
-                      ? strings.drefExportImminentWhereWhenHow
-                      : strings.drefExportWhatWhereWhen}
+                    {!isImminentOnset ?
+                      strings.drefFormWhatWhereWhen
```
Comment (AdityaKhatri) [inline src/root/components/OperationalUpdatePdfDocument/index.tsx:514]:
```suggestion
                    {!isImminentOnset
                      ? strings.drefFormWhatWhereWhen
```

File: src/root/components/OperationalUpdatePdfDocument/index.tsx [hunk 4]
```diff
@@ -732,9 +712,9 @@ function OperationalUpdatePdfDocument(props: Props) {
         {operationalUpdateResponse.needs_identified.length > 0 && (
           <View style={pdfStyles.niSection} wrap={false}>
             <Text style={pdfStyles.sectionHeading}>
-              {isImminentOnset
-                ? strings.drefExportImminentNeedsGapsIdentified
-                : strings.drefExportNeedsGapsIdentified}
+              {isImminentOnset ?
+                strings.drefFormImminentNeedsIdentified
```
Comment (AdityaKhatri) [inline src/root/components/OperationalUpdatePdfDocument/index.tsx:716]:
```suggestion
              {isImminentOnset
               ? strings.drefFormImminentNeedsIdentified
```

File: src/root/views/DrefApplicationForm/DrefOverview/CountryDistrictInput/index.tsx [hunk 5]
```diff
@@ -1,117 +1,105 @@
 import React from 'react';
-import {
-  PartialForm,
-  ArrayError,
-  useFormObject,
-  getErrorObject,
-} from '@togglecorp/toggle-form';
-import { IoTrash } from 'react-icons/io5';
+//import {
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/DrefOverview/CountryDistrictInput/index.tsx:2]: Why is this commented out?

File: src/root/views/DrefApplicationForm/EventDetails/index.tsx [hunk 6]
```diff
@@ -14,12 +16,15 @@ import RadioInput from '#components/RadioInput';
 import TextArea from '#components/TextArea';
 import LanguageContext from '#root/languageContext';
 import DREFFileInput from '#components/DREFFileInput';
+import DateInput from '#components/DateInput';
 
+import CaptionInput from '../CaptionInput/CaptionInput';
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/EventDetails/index.tsx:21]: Why is there 2 CaptionInput folders?

File: src/root/views/DrefApplicationForm/EventDetails/index.tsx [hunk 7]
```diff
@@ -49,6 +54,32 @@ function EventDetails(props: Props) {
   } = props;
 
   const error = getErrorObject(formError);
+  const imagesValue = React.useMemo(() => (
+    value?.images_file?.map(d => d.id).filter(d => !!d) as number[] | undefined
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/EventDetails/index.tsx:58]: Lets not cast the output. Is image_files not properly typed?

File: src/root/views/DrefApplicationForm/EventDetails/index.tsx [hunk 8]
```diff
@@ -184,20 +244,35 @@ function EventDetails(props: Props) {
...
+              <CaptionInput
+                key={g.client_id}
+                index={i}
+                value={g}
+                onChange={onImageChange}
+                onRemove={onImageRemove}
+                error={getErrorObject(error?.images_file)}
+                fileIdToUrlMap={fileIdToUrlMap}
+              />
+            ))}
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/EventDetails/index.tsx:274]: Lets use list

File: src/root/views/DrefApplicationForm/Response/RiskSecurityInput/index.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+import { IoTrash } from 'react-icons/io5';
+
+import Button from '#components/Button';
+import LanguageContext from '#root/languageContext';
+import { RiskSecurityType } from '#views/DrefApplicationForm/useDrefFormOptions';
+import TextInput from '#components/TextInput';
+
+import styles from './styles.module.scss';
+
+type SetValueArg<T> = T | ((value: T) => T);
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/Response/RiskSecurityInput/index.tsx:17]: Lets define this in utils and use it here

File: src/root/views/DrefApplicationForm/Response/RiskSecurityInput/index.tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+const defaultCountryDistrictValue: PartialForm<RiskSecurityType> = {
+  clientId: 'test',
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/Response/RiskSecurityInput/index.tsx:20]: use randomString()

File: src/root/views/DrefApplicationForm/index.tsx [hunk 11]
```diff
@@ -448,6 +457,63 @@ function DrefApplication(props: Props) {
...
+    const dynamicTitle = `${countryName} ${filteredDisasterTypeName} ${value?.event_date ?? ''}`;
+    onValueChange(dynamicTitle, 'title');
+  }, [
+    countryOptions,
+    disasterTypeOptions,
+    value.disaster_type,
+    value.country,
+    value.event_date,
+    onValueChange,
+  ]);
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/index.tsx:475]: Not sure how this works.

File: src/root/views/DrefApplicationForm/Response/index.tsx [hunk 12]
```diff
@@ -125,17 +136,80 @@ function Response(props: Props) {
...
+    onValueChange,
+  );
+
+  type riskSecurity = typeof value.risk_security;
+
+  const handleRiskSecurityAdd = React.useCallback(() => {
+    const clientId = randomString();
+    const newList: PartialForm<RiskSecurityType> = {
+      clientId,
+    };
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/Response/index.tsx:166]:
```suggestion
    const newList: PartialForm<RiskSecurityType> = {
      clientId: randomString(),
    };
```

File: src/root/views/DrefApplicationForm/Response/index.tsx [hunk 13]
```diff
@@ -273,32 +337,96 @@ function Response(props: Props) {
...
-          />
+          {value.risk_security?.map((rs, i) => (
+            <RiskSecurityInput
+              key={rs.clientId}
+              index={i}
+              value={rs}
+              onChange={onRiskSecurityChange}
+              onRemove={onRiskSecurityRemove}
+              error={getErrorObject(error?.risk_security)}
+            />
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/Response/index.tsx:356]: Use list

File: src/root/views/DrefApplicationForm/Response/index.tsx [hunk 14]
```diff
@@ -273,32 +337,96 @@ function Response(props: Props) {
...
+          <InterventionInput
+            key={n.clientId}
+            index={i}
+            value={n}
+            onChange={onInterventionChange}
+            onRemove={onInterventionRemove}
+            error={getErrorObject(error?.planned_interventions)}
+            interventionOptions={interventionOptions}
+            showNewFieldOperational={false} />
+        ))}
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/Response/index.tsx:428]: use list

File: src/root/views/DrefApplicationForm/DrefOverview/index.tsx [hunk 15]
```diff
@@ -133,21 +112,81 @@ function DrefOverview(props: Props) {
...
+  ), [userOptions]);
+
   const initialOptions = React.useMemo(() => (
     value.users?.map((u) => ({
       label: userMap[u],
       value: u,
     }))
   ), [userMap, value.users]);
 
+  const handleNSChange = useCallback((ns) => {
```
Comment (AdityaKhatri) [inline src/root/views/DrefApplicationForm/DrefOverview/index.tsx:128]: define type of ns

File: src/root/views/DrefApplicationForm/EventDetails/index.tsx (hunk 7 above)
Comment (frozenhelium) [reply src/root/views/DrefApplicationForm/EventDetails/index.tsx:58]: Looking at the code it looks like, the transformation here is from `(number | undefined)[]` to `number[]` which is achieved from the `filter`

**Pull request merged**